kahuna/public/js/usage-rights/usage-rights-editor.js

```diff
@@ -6,125 +6,112 @@ import './usage-rights-editor.css!';
 
 import '../components/gr-confirm-delete/gr-confirm-delete.js';
 
+
+import Rx from 'rx';
+import '../util/rx';
+
 export var usageRightsEditor = angular.module('kahuna.edits.usageRightsEditor', [
     'monospaced.elastic',
-    'gr.confirmDelete'
+    'gr.confirmDelete',
+    'util.rx'
 ]);
 
 usageRightsEditor.controller(
     'UsageRightsEditorCtrl',
-    ['$q', '$scope', '$window', '$timeout', 'editsService', 'editsApi', 'onValChange',
-    function($q, $scope, $window, $timeout, editsService, editsApi, onValChange) {
+    ['$q', '$scope', 'inject$', 'observe$', 'editsService', 'editsApi',
+    function($q, $scope, inject$, observe$, editsService, editsApi) {
 
     var ctrl = this;
+    const multiCat = { name: 'Multiple categories', value: 'multi-cat', properties: [] };
 
-    ctrl.saving = false;
-    ctrl.saved = false;
-    ctrl.categories = [];
-    ctrl.originalCats = [];
-    ctrl.model = angular.extend({}, ctrl.usageRights.data);
-
-    const multiRights = { name: 'Multiple categories', value: '' };
-    const noRights = { name: 'None', value: '' };
-    const catsWithNoRights = () => [noRights].concat(ctrl.originalCats);
-    const catsWithMultiRights = () => [multiRights].concat(ctrl.originalCats);
-    const setStandardCats = () => ctrl.categories = ctrl.originalCats;
-    const setNoRightsCats = () => {
-        ctrl.categories = catsWithNoRights();
-        ctrl.category = noRights;
-    };
-    const setMultiRightsCats = () => {
-        ctrl.categories = catsWithMultiRights();
-        ctrl.category = multiRights;
-    };
-
+    // @return Stream.<Array.<UsageRights>>
+    const usageRights$ = observe$($scope, () => ctrl.usageRights).startWith([]);
```

Comment:
`$watch` can take a string. The string has to be an expression that evaluates on the scope though, so not some custom function, and presumably because we sometimes needed the freedom of custom functions, we also used functions even when a string would have worked.

I don't mind too much either way, just pointing it out.